hw/ip_templates/pwrmgr/util/ipconfig.py

```diff
@@ -0,0 +1,53 @@
+# Copyright lowRISC contributors (OpenTitan project).
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+"""
+This contains a class to access the pwrmgr's configuration from its ipconfig
+files.
+"""
+from typing import List, Dict
+
+
+class IpConfig:
+    def __init__(self, ipconfig: object):
+        """
+        Initialize an `IpConfig` from an already loaded and parsed `ipconfog.hjson`
+        files.
+        """
+        self.param_values = ipconfig["param_values"]
+
+    def wakeup_list(self) -> List[Dict]:
```

Comment:
For ipgen we will want iterables rather than collections.

Reply:
Sorry for the delay. For ipgen we tpically iterate, so returning iterators would be marginally better.